From 0ecfea397a0bb710458262b5553e2faaa7a1be0b Mon Sep 17 00:00:00 2001
Message-Id: <0ecfea397a0bb710458262b5553e2faaa7a1be0b.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Pallavi <pallavi.g@intel.com>
Date: Mon, 25 Aug 2014 11:27:07 +0530
Subject: [PATCH 20/75] MUST_REBASE [VPG]: adf/intel: Add new plane capability
 support in core intel adf

This can be used by GEN devices that support these new capabilities

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I5b44162983976d9099ed8e1bdfb5a74071c4b1ec
Signed-off-by: Pallavi <pallavi.g@intel.com>
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 .../video/adf/intel/include/core/intel_dc_config.h |   29 ++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/drivers/video/adf/intel/include/core/intel_dc_config.h b/drivers/video/adf/intel/include/core/intel_dc_config.h
index cc423dc..d857f25 100644
--- a/drivers/video/adf/intel/include/core/intel_dc_config.h
+++ b/drivers/video/adf/intel/include/core/intel_dc_config.h
@@ -54,6 +54,26 @@ enum intel_plane_compression {
 	INTEL_PLANE_DECOMPRESSION_16X4,
 };
 
+enum intel_plane_tiling_mode {
+	INTEL_PLANE_TILE_NONE,
+	INTEL_PLANE_TILE_X,
+	INTEL_PLANE_TILE_Y,
+};
+
+enum intel_plane_reserved_bit {
+	INTEL_PLANE_RESERVED_BIT_ZERO,
+	INTEL_PLANE_RESERVED_BIT_SET,
+};
+
+enum intel_plane_zorder {
+	INTEL_PLANE_P1S1S2C1,
+	INTEL_PLANE_P1S2S1C1,
+	INTEL_PLANE_S2P1S1C1,
+	INTEL_PLANE_S2S1P1C1,
+	INTEL_PLANE_S1P1S2C1,
+	INTEL_PLANE_S1S2P1C1,
+};
+
 struct intel_plane_capabilities {
 	const u32 *supported_formats;
 	const size_t n_supported_formats;
@@ -69,6 +89,15 @@ struct intel_plane_capabilities {
 
 	const u32 *supported_decompressions;
 	const size_t n_supported_decompressions;
+
+	const u32 *supported_tiling;
+	const size_t n_supported_tiling;
+
+	const u32 *supported_zorder;
+	const size_t n_supported_zorder;
+
+	const u32 *supported_reservedbit;
+	const size_t n_supported_reservedbit;
 };
 
 struct intel_buffer {
-- 
1.7.9.5

